### ISSUE TITLE
Unable to assign multiple identities using a single selector

### ISSUE THREAD

**Issue by GlorifiedTypist** [Azure/aad-pod-identity#1048]
**Have you**

- [x] Read [Troubleshooting Guide](../../docs/readmes/README.troubleshooting.md)
- [x] Read [Known Issues](../../docs/readmes/README.known-issues.md)
- [x] Searched on [GitHub issues](https://github.com/Azure/aad-pod-identity/issues)

**Describe the bug**
According to the documentation [here](https://github.com/Azure/aad-pod-identity/blob/e95ed2767c032805773eb78b84b5a48ee6844148/docs/readmes/README.best-practices.md#a-pod-using-multiple-azureidentities) pods should be able to assume more than one identity using a single selector. In practise only the first identity is assumed.

**Steps To Reproduce**
Create multiple identities referenced by a single shared selector as per doc [here](https://github.com/Azure/aad-pod-identity/blob/e95ed2767c032805773eb78b84b5a48ee6844148/docs/readmes/README.best-practices.md#a-pod-using-multiple-azureidentities)

**AzureIdentities**

```
apiVersion: aadpodidentity.k8s.io/v1
kind: AzureIdentity
metadata:
  name: dev-shared-app
  namespace: dev
spec:
  clientID: 00000000-0000-0000-0000-0000-00000001
  resourceID: /subscriptions/00000000-0000-0000-0000-0000-00000000/resourceGroups/my-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/identity01
  type: 0
```

```
apiVersion: aadpodidentity.k8s.io/v1
kind: AzureIdentity
metadata:
  name: dev-ops-shared-app
  namespace: dev
spec:
  clientID: 00000000-0000-0000-0000-0000-00000002
  resourceID: /subscriptions/00000000-0000-0000-0000-0000-00000000/resourceGroups/my-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/identity02
  type: 0
```

**AzureIdentityBindings**

```
apiVersion: aadpodidentity.k8s.io/v1
kind: AzureIdentityBinding
metadata:
  name: dev-ops-shared-app-binding
  namespace: dev
spec:
  azureIdentity: dev-ops-shared-app
  selector: dev-shared-app
```

```
apiVersion: aadpodidentity.k8s.io/v1
kind: AzureIdentityBinding
metadata:
  name: dev-shared-app-binding
  namespace: dev
spec:
  azureIdentity: dev-shared-app
  selector: dev-shared-app
```

**Deployment**

```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: vault-test
    aadpodidbinding: dev-shared-app
  name: vault-test
  namespace: dev
spec:
  replicas: 1
  selector:
    matchLabels:
      app: vault-test
      aadpodidbinding: dev-shared-app
  strategy: {}
  template:
    metadata:
      labels:
        app: vault-test
        aadpodidbinding: dev-shared-app
    spec:
      containers:
        - image: alpine
          command:
            - sleep
            - "93600"
          name: alpine
          resources: {}
```

When applying identities and binding, logs:

I0415 14:12:03.371405       1 mic.go:589] pod dev/vault-test-75cbbc9b46-7xq78 has no assigned node yet. it will be ignored
I0415 14:12:03.375295       1 mic.go:1027] processing node node-000001, add [2], del [0], update [0]
I0415 14:12:03.375328       1 crd.go:513] creating assigned id dev/vault-test-75cbbc9b46-7xq78-dev-dev-shared-app
I0415 14:12:03.375353       1 crd.go:513] creating assigned id dev/vault-test-75cbbc9b46-7xq78-dev-dev-ops-shared-app
I0415 14:12:03.447030       1 cloudprovider.go:215] updating user-assigned identities on node-000001, assign [2], unassign [0]
I0415 14:12:16.904645       1 crd.go:776] updating AzureAssignedIdentity dev/vault-test-75cbbc9b46-7xq78-dev-dev-shared-app status to Assigned
I0415 14:12:16.904717       1 crd.go:776] updating AzureAssignedIdentity dev/vault-test-75cbbc9b46-7xq78-dev-dev-ops-shared-app status to Assigned

When fetching a token from within the deployed pod, the first identity is matched:

`TOKEN=$(curl -v 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://vault.azure.net' -H Metadata:true | awk -F
"[{,\":}]" '{print $6}')`

I0415 14:13:25.062326       1 standard.go:73] no clientID or resourceID in request. dev/vault-test-75cbbc9b46-7xq78 has been matched with azure identity dev/dev-shared-app
I0415 14:13:25.062388       1 standard.go:179] matched identityType:0 clientid:45bf##### REDACTED #####a022 resource:https://vault.azure.net
I0415 14:13:28.851345       1 server.go:196] status (200) took 3789118951 ns for req.method=GET reg.path=/metadata/identity/oauth2/token req.remote=10.101.64.79

As a result the pod can only assume permissions of the first identity `dev-shared-app`.

I can query the vault with the identity `dev-shared-app` attached but cannot query the second vault with the identity `dev-ops-shared-app` attached due to authentication failure.

**Expected behavior**
Able to assume both identities and sucessfully query both vaults (in this case) where the identities are attached.

**AAD Pod Identity version**
v1.7.5. Build date: 2021-04-02-21:14

**Kubernetes version**
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.5", GitCommit:"e6503f8d8f769ace2f338794c914a96fc335df0f", GitTreeState:"clean", BuildDate:"2020-06-26T03:47:41Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.14", GitCommit:"5de7fd1f9555368a86eb0f8f664dc58055c17269", GitTreeState:"clean", BuildDate:"2021-01-18T09:31:01Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

**Additional context**

**Comment by aramase** [comment]
@GlorifiedTypist Using the same selector for multiple identity will give the application pod access to multiple identities. However the application is required to pass the clientID for the identity it wants a token for. By default without aad-pod-identity if a token request is made using curl without the identity, then IMDS will return an error because it doesn't know which identity to pick. However with pod identity, we provide a silent feature where if the client doesn't set the clientID in token request, we set it on behalf of the application pod with the clientID from the first `AzureIdentity` matched for the pod.

tl;dr - If you're binding multiple identities to the pod, you would need to specify the clientID in the token request so that aad-pod-identity/IMDS can pick that identity for you and fetch a token.

**Comment by GlorifiedTypist** [comment]
This makes perfect sense. I have tested this and its working as expected now. Can we add a one liner to the documentation to clarify?

**Comment by aramase** [comment]
Sure! Would you be interested in opening a PR to update the doc?
